platform-sdk/swirlds-state-api/src/testFixtures/java/com/swirlds/state/test/fixtures/merkle/TestStringRecordCodec.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright (C) 2023-2024 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.swirlds.state.test.fixtures.merkle;
+
+import com.hedera.pbj.runtime.Codec;
+import com.hedera.pbj.runtime.ParseException;
+import com.hedera.pbj.runtime.io.ReadableSequentialData;
+import com.hedera.pbj.runtime.io.WritableSequentialData;
+import com.swirlds.state.test.fixtures.StringRecord;
+import edu.umd.cs.findbugs.annotations.NonNull;
+
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.util.Objects;
+
+/**
+ * This is a test-only class used exclusively by MerkleTestBase.
+ * <strong>This class should not be used elsewhere.</strong>
+ * The use of Codec implementations is not ideal, and we have the protobuf ProtoString as
+ * a replacement.  When time permits this class needs to be removed and all usages replaced
+ * with ProtoString and the ProtoString.PROTOBUF codec.
+ * @deprecated Use ProtoString and ProtoString.PROTOBUF instead of String and this codec.
+ */
+@Deprecated
+@SuppressWarnings("Singleton")
+public class TestStringRecordCodec implements Codec<StringRecord> {
+    public static final TestStringRecordCodec SINGLETON = new TestStringRecordCodec();
+
+    private TestStringRecordCodec() {}
+
+    @NonNull
+    @Override
+    public StringRecord parse(@NonNull ReadableSequentialData input, boolean strictMode, int maxDepth) throws ParseException {
```

Comment:
All of these parameters _could_ be declared `final`.  The general policy for this codebase is that any method parameter that _could_ be `final`, _should_ be declared as such.
Doing so in all of the newly added code will resolve most of the current static analysis issues.

Reply:
Done